### ISSUE TITLE
Fix Cast tests on Databricks 14.3

### ISSUE THREAD

**Issue by razajafri** [NVIDIA/spark-rapids#11530]
Build the plugin against the Databricks 14.3 cluster using #11467. Once built successfully run the cast tests by `TESTS=cast_test.py jenkins/databricks/test.sh` 

The following tests fail 
```
[gw2] [ 20%] FAILED ../../src/main/python/cast_test.py::test_cast_timestamp_to_numeric_non_ansi
```

**Comment by mythrocks** [comment]
I'm taking a look at this one.  I've seen this failure in the context of Spark 4.

**Comment by mythrocks** [comment]
https://github.com/NVIDIA/spark-rapids/pull/11559 should now address this too.

**Comment by mythrocks** [comment]
This is addressed by #11559.
